### DISCUSSION TITLE
Connection error: Proxmox 8.2.2 & TrueNAS Scale 24.04

### DISCUSSION THREAD

**Discussion by jamesstanw** [TheGrandWazoo/freenas-proxmox#196] [Q&A] [answered]
Hopefully I've not missed the obvious.  With the current setup, and testing as below, I have the following error.  Hopefully someone can help me get off this rock . . . 

`command '/usr/bin/ssh -o 'BatchMode=yes' -i /etc/pve/priv/zfs/10.0.0.1_id_rsa root@10.0.0.1 zfs list -o name,volsize,origin,type,refquota -t volume,filesystem -d1 -Hp Web-storage-pool' failed: exit code 255 (500)`

- TrueNAS (TN) ip is 10.0.0.1/30 and Proxmox (PM) is 10.0.0.2/30 - they respond to ping from both sides
- SSH keys are configured.  I can login to TN from PM and execute commands (e.g., mkdir)
- On the PM box I'm not sure the API is working correctly?  A curl command shows this:

root@nodes:~# curl -k -l https://10.0.0.1/api/
<html>
<head><title>302 Found</title></head>
<body>
<center><h1>302 Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
root@nodes:~#

- My storage.cfg shows:

`zfs: NAS4-web
        blocksize 4k
        iscsiprovider freenas
        pool Web-storage-pool
        portal 10.0.0.1
        target iqn.[default on system]:web-store
        content images
        freenas_apiv4_host 10.0.0.1
        freenas_password *************** [removed but NAS4 root pw]
        freenas_use_ssl 0
        freenas_user root
        nowritecache 0
        sparse 1
`
- on the TN side,  Web-storage-pool is, just a pool (mirror) of disks and shows up as:
`root@truenas[~]# zfs list -o name,volsize,origin,type,refquota -t volume,filesystem -d1 -Hp Web-storage-pool

Web-storage-pool        -       -       filesystem      0
Web-storage-pool/.system        -       -       filesystem      0
`
Alternatively:
`root@truenas[~]# zpool status                                                                               
  pool: Web-storage-pool
 state: ONLINE
config:

        NAME                                      STATE     READ WRITE CKSUM
        Web-storage-pool                          ONLINE       0     0     0
          mirror-0                                ONLINE       0     0     0
            5825fef5-7518-4a35-bf83-7a613802572a  ONLINE       0     0     0
            9819ac68-fc74-463a-9314-dc63ef10e331  ONLINE       0     0     0
            d01b92dc-d868-4e3f-8743-7eae5268a31a  ONLINE       0     0     0
            be4853df-5470-4b3f-a723-c2f2423f1a38  ONLINE       0     0     0

errors: No known data errors

  pool: boot-pool
 state: ONLINE
config:

        NAME         STATE     READ WRITE CKSUM
        boot-pool    ONLINE       0     0     0
          nvme2n1p3  ONLINE       0     0     0

errors: No known data errors`

On the TN side the sharing-iSCSI settings are:
Target Global Configuration - default as system defaults
Portals - Group ID = 2 Listen = 10.0.0.1:3260 Description = iscsi-portal-t Discovery Auth Method = NONE
Initiators Groups - Group ID =2 Initiators = iqn******* from PM Description = prox-box
Authorized Access = no records added yet
Targets - Target Name = web-store Network = 10.0.0.0/30 Portal Group ID = 1(iscsi-portal-t) Initiator Group ID = 2(iqn.*******) Authentication Method = None
Extents = No records added yet
Associated Targets = No records added yet

**Comment by TheGrandWazoo** [comment, marked as answer]
Apologies for not following up here. This is the old SSH + `zfs` CLI driver (`LunCmd/FreeNAS.pm`, the `freenas_apiv4_host`-style config) — that whole approach was replaced in v3.0, which talks to TrueNAS purely over the REST API with a bearer token, no SSH keys required. Given this is from over a year ago I can't debug the exact SSH failure now, but if you're still running into connectivity issues I'd recommend moving to v3.0 rather than chasing the SSH path — it removes this entire class of problem. See the [README](README.md) Quick Start and [Migrating from v2.x](docs/migrating-from-v2.md).

**Comment by jamesstanw** [comment]
Thanks for the reply!
In the end I made this work with regular iSCSI storage. . . then wondered
why Proxmox would not let me create snapshots etc. Generally this has
worked fine for me - though it can be a little troublesome to have to work
both sides e.g., for an occasional storage resize.
At this point could I remove and then re-add these volumes as TrueNAS
(ZFS/iSCSI) shares?  Am I going to break stuff doing this?  The ability to
create snapshots would be super useful now when I'm migrating Nextcloud
across PHP versions.
